### PR TITLE
PMM-8082 srv init

### PR DESCRIPTION
**Pull request by nikita-b**
https://github.com/Percona-Lab/pmm-submodules/pull/2534
https://github.com/percona/pmm-update/pull/215
https://github.com/percona/pmm-server/pull/364
https://github.com/percona/pmm-managed/pull/1148

Build: SUBMODULES-2534

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,7 +1,21 @@
 #!/bin/bash
-
 set -o errexit
 
+# init /srv if empty
+DIST_FILE=/srv/pmm-distribution
+if [ ! -f $DIST_FILE ]; then
+    echo "File $DIST_FILE doesn't exist. Initizlize /srv..."
+    echo docker > $DIST_FILE
+    mkdir -p /srv/{clickhouse,grafana,logs,postgres,prometheus,nginx,victoriametrics}
```
Comment (atymchuk) [inline entrypoint.sh:9]: I think, we are migrating `postgres` to contain the version, i.e. `postgres14` if I look at this https://github.com/percona/pmm-server/pull/362/files#diff-500a0e8dcb2879d903640ba46031362cdc923d030e278cf2a681c24765cc87b0R41

Comment (atymchuk) [reply entrypoint.sh:9]: OK if this will be merged prior to that migration PR.

**Review by atymchuk — APPROVED** [review]

File: entrypoint.sh [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+    chown grafana:grafana /srv/grafana
+    chown pmm:pmm /srv/{victoriametrics,prometheus,logs}
+    chown postgres:postgres /srv/postgres
+    echo "Generate self-signed certificates for nginx"
+    bash /var/lib/cloud/scripts/per-boot/generate-ssl-certificate
+    echo "Init Postgres"
+    su postgres -c "/usr/pgsql-11/bin/initdb -D /srv/postgres"
+fi
```
Comment (denisok) [inline entrypoint.sh:17]: would be good to have PMM admin password here as well. As this is an entry point, probably some more standard env name should be chosen, or probably mount point with a pass file ?

https://github.com/percona/percona-helm-charts/pull/132/files#diff-adad060883d938d361b0636b890c5e0a39ff6a9670b6ec9d06d04165293e1019R103

File: entrypoint.sh [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+DIST_FILE=/srv/pmm-distribution
+if [ ! -f $DIST_FILE ]; then
+    echo "File $DIST_FILE doesn't exist. Initizlize /srv..."
+    echo docker > $DIST_FILE
+    mkdir -p /srv/{clickhouse,grafana,logs,postgres,prometheus,nginx,victoriametrics}
+    chown grafana:grafana /srv/grafana
+    chown pmm:pmm /srv/{victoriametrics,prometheus,logs}
+    chown postgres:postgres /srv/postgres
+    echo "Generate self-signed certificates for nginx"
+    bash /var/lib/cloud/scripts/per-boot/generate-ssl-certificate
```
Comment (denisok) [inline entrypoint.sh:14]: yeah, this could take some time. In air gaped env this will fail. I think we had PoC that we have copied it from /etc.

**Review by denisok — APPROVED** [review]

**Review by denisok — APPROVED** [review]

File: entrypoint.sh [hunk 4]
```diff
@@ -1,7 +1,32 @@
...
+    echo "Init Postgres"
+    su postgres -c "/usr/pgsql-11/bin/initdb -D /srv/postgres"
+    echo "Temporary start postgres and enable pg_stat_statements"
+    su postgres -c "/usr/pgsql-11/bin/pg_ctl start -D /srv/postgres"
+    su postgres -c "psql postgres postgres -c 'CREATE EXTENSION pg_stat_statements SCHEMA public'"
+    su postgres -c "/usr/pgsql-11/bin/pg_ctl stop -D /srv/postgres"
+fi
+
+if [ -n "$PMM_ADMIN_PASSWORD" ]; then
```
Comment (denisok) [inline entrypoint.sh:25]: I am thinking if that also belongs to `init` only?

**Review by denisok — APPROVED** [review]

File: entrypoint.sh (hunk 4 above)
Comment (nikita-b) [reply entrypoint.sh]: (discussed it in ticket)

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**